akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterShardingSettings.scala

```diff
@@ -38,7 +38,10 @@ object ClusterShardingSettings {
       leastShardAllocationMaxSimultaneousRebalance =
         config.getInt("least-shard-allocation-strategy.max-simultaneous-rebalance"),
       waitingForStateTimeout = config.getDuration("waiting-for-state-timeout", MILLISECONDS).millis,
-      updatingStateTimeout = config.getDuration("updating-state-timeout", MILLISECONDS).millis)
+      updatingStateTimeout = config.getDuration("updating-state-timeout", MILLISECONDS).millis,
+      entityRecoveryStrategy = config.getString("entity-recovery-strategy"),
```

Comment:
If you look at line 94 and 95 the field is validated so it shouldn't fail later on in the actor. This follows the same approach as is done for stateStoreMode in ClusterShardingSettings so keeping it consistent. Either way I'm happy to change this to use a more typed approach if you feel that this a better approach. 

In terms of just passing the class names... this was the approach taken in early revisions of this PR (i.e. a more pluggable setup) which was thought too much at this stage.